### PR TITLE
Use block height for AtomicSwapEther timeout

### LINKED ISSUE #14: Convert atomic swap timeout from unix timestamp to block height

### PR DESCRIPTION
**Pull request by willclarktech**
Closes #14 

Suddenly got issues with number comparisons, not sure why they didn't pop up before.

### CODE REVIEW

File: test/AtomicSwapEther.js [hunk 1]
```diff
@@ -1,13 +1,15 @@
...
+const { getBalance, makeRandomAddress, makeRandomId, makeTimeout, sleep } = require("./utils");
 
 const atomicSwap = artifacts.require("./AtomicSwapEther.sol");
 
 contract("AtomicSwapEther", accounts => {
   const defaultPreimage = "0x42a990655bffe188c9823a2f914641a32dcbb1b28e8586bd29af291db7dcd4e8";
   const defaultHash = "0x261c74f7dd1ed6a069e18375ab2bee9afcb1095613f53b07de11829ac66cdfcc";
   const nullPreimage = "0x0000000000000000000000000000000000000000000000000000000000000000";
-  const defaultAmount = 50e6;
+  const defaultAmount = '50000000';
```
Comment (webmaster128) [inline test/AtomicSwapEther.js]: Not linted

File: test/utils.js [hunk 2]
```diff
@@ -18,17 +19,18 @@ function makeRandomAddress() {
   return toChecksummedAddress(addressLowercase);
 }
 
-function makeTimeout(seconds = 100) {
-  const currentUnixTimestamp = Math.floor(Date.now() / 1000);
-  return currentUnixTimestamp + seconds;
+async function makeTimeout(blocks = 2) {
+  const connection = await EthereumConnection.establish("http://localhost:8545");
```
Comment (webmaster128) [inline test/utils.js]: Can we dedup the URL?

File: test/utils.js [hunk 3]
```diff
@@ -1,12 +1,13 @@
...
 
-async function getBalanceApproximation(address) {
-  const connection = await EthereumConnection.establish("http://localhost:7545");
+async function getBalance(address) {
+  const connection = await EthereumConnection.establish("http://localhost:8545");
   const account = await connection.getAccount({ address });
   const ethWallet = account && account.balance.find(({ tokenTicker }) => tokenTicker === "ETH");
   const balance = ethWallet ? ethWallet.quantity : "0";
-  return parseInt(balance, 10);
+  return new BN(balance, 10);
```
Comment (webmaster128) [inline test/utils.js:17]: We should call `connection.disconnect()` here before returning, just to prepare for upcoming websocket implementation.

File: test/utils.js [hunk 4]
```diff
@@ -18,17 +19,18 @@ function makeRandomAddress() {
   return toChecksummedAddress(addressLowercase);
 }
 
-function makeTimeout(seconds = 100) {
-  const currentUnixTimestamp = Math.floor(Date.now() / 1000);
-  return currentUnixTimestamp + seconds;
+async function makeTimeout(blocks = 2) {
+  const connection = await EthereumConnection.establish("http://localhost:8545");
+  const currentHeight = await connection.height();
+  return currentHeight + blocks;
```
Comment (webmaster128) [inline test/utils.js:34]: `connection.disconnect()`

File: truffle-config.js [hunk 5]
```diff
@@ -43,11 +43,12 @@ module.exports = {
...
-    // development: {
-    //  host: "127.0.0.1",     // Localhost (default: none)
-    //  port: 8545,            // Standard Ethereum port (default: none)
-    //  network_id: "*",       // Any network (default: none)
-    // },
+    development: {
+     host: "127.0.0.1",     // Localhost (default: none)
+     port: 8545,            // Standard Ethereum port (default: none)
+     network_id: "*",       // Any network (default: none)
+     blockTime: 1,          // Default is 0 (a block is created when a transaction is received)
```
Comment (webmaster128) [inline truffle-config.js]: Can we use a different port than 8545, which we use for the IOV-Core ganache that has a different configuration? https://github.com/iov-one/iov-core/blob/master/scripts/ethereum/start.sh
This is running almost all the time on my machine.

Comment (webmaster128) [reply truffle-config.js]: Isn't ganache-cli started from scripts/test.sh without those settings?

Comment (willclarktech) [reply truffle-config.js]: It is, but truffle uses them to know which network to connect. I'll update.

Comment (webmaster128) [reply truffle-config.js]: But than `blockTime: 1` is the wrong value as ganache-cli defaults to 0 (_a block is created when a transaction is received_)

Comment (willclarktech) [reply truffle-config.js]: Yep, the truffle client doesn't need the block time.

Comment (webmaster128) [reply truffle-config.js]: okay, so it should either be unset or set to 0, right?

**Review by webmaster128 — APPROVED** [review]
👍

**Pull request merged**